src/main/java/org/broadinstitute/hellbender/metrics/MultiLevelCollectorReducible.java

```diff
@@ -0,0 +1,428 @@
+            super(rgRecs);
+        }
+
+        @Override
+        protected String getKey(SAMReadGroupRecord rg) {
+            return rg.getLibrary();
+        }
+
+        @Override
+        protected UNIT_COLLECTOR makeCollector(SAMReadGroupRecord rg) {
+            return makeLibraryCollector(rg);
+        }
+
+        @Override
+        protected UNIT_COLLECTOR makeUnknownCollector() {
+            return makeChildCollector(UNKNOWN, UNKNOWN, null);
+        }
+    }
+
+    //Discriminates between records based on read group name, and calls acceptRecord on the appropriate PerUnitMetricCollectors
+    private class ReadGroupDistributor extends Distributor implements Serializable {
+        private static final long serialVersionUID = 1L;
+
+        public ReadGroupDistributor(final List<SAMReadGroupRecord> rgRecs) {
+            super(rgRecs);
+        }
+
+        @Override
+        protected String getKey(SAMReadGroupRecord rg) {
+            return rg.getPlatformUnit();
+        }
+
+        @Override
+        protected UNIT_COLLECTOR makeCollector(SAMReadGroupRecord rg) {
+            return makeReadGroupCollector(rg);
+        }
+
+        @Override
+        protected UNIT_COLLECTOR makeUnknownCollector() {
+            return makeChildCollector(UNKNOWN, UNKNOWN, UNKNOWN);
+        }
+    }
+
+    /**
+     * Use an init method so that overloaded methods in subclasses can pass use parameters that are initialized in their constructor
+     * @param accumulationLevels PerUnitMetricCollectors will only be created for the levels identified by accumulationLevels
+     * @param samRgRecords PerUnitMetricCollectors will be created for each of the different samples, libraries, and
+     *                     readGroups found in the records depending on the accumulationLevels provided
+     */
+    protected void setup(final Set<MetricAccumulationLevel> accumulationLevels, final List<SAMReadGroupRecord> samRgRecords) {
+        outputOrderedDistributors = new ArrayList<>(4);
+        if(accumulationLevels.contains(MetricAccumulationLevel.ALL_READS)) {
+            outputOrderedDistributors.add(new AllReadsDistributor(samRgRecords));
+        }
+        if (accumulationLevels.contains(MetricAccumulationLevel.SAMPLE)) {
+            outputOrderedDistributors.add(new SampleDistributor(samRgRecords));
+        }
+
+        if(accumulationLevels.contains(MetricAccumulationLevel.LIBRARY)) {
+            outputOrderedDistributors.add(new LibraryDistributor(samRgRecords));
+        }
+
+        if(accumulationLevels.contains(MetricAccumulationLevel.READ_GROUP)) {
+            outputOrderedDistributors.add(new ReadGroupDistributor(samRgRecords));
+        }
+    }
+
+    /**
+     * Construct a argument of ARGTYPE using the given SAMRecord and ReferenceSequence then pass
+     * this value to all collectors that should include this record
+     */
+    public void acceptRecord(final SAMRecord record, final ReferenceSequence refSeq) {
```

Comment:
This one isn't an override in the Distributor hierarchy; its the original in the MultiLevelCollector hierarchy.